src/promise.js

```diff
@@ -753,6 +753,14 @@ Promise.prototype._settledValue = function() {
     // Implicit undefined for cancelled promise.
 };
 
+if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
+    es5.defineProperty(Promise.prototype, Symbol.toStringTag, {
+        get: function () {
+            return "Promise";
```

Comment:
If the value returned is `"Object"` instead, we would get the exact same external `toString.call` behavior as we do now

Reply:
Additionally, libraries that use x[Symbol.toStringTag] === 'Promise' to detect promises will continue to work normally. Only libraries that check for the mere presence of `x[Symbol.toStringTag]` will have problems.

Hmm. I see [we'd still have an incompatible type though](https://github.com/Microsoft/TypeScript/blob/61fb222cd2fc79ea30aabbc818a05550f6e5ac83/lib/lib.es2015.symbol.wellknown.d.ts#L168-L170)

Reply:
Wrote about it here: https://github.com/Microsoft/TypeScript/issues/19006#issuecomment-396874310